stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/ReadCharacters.java

```diff
@@ -55,29 +65,81 @@
 
+    /**
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(ReadCharacters.class);
+
+    /*
+     * Callback method of the read characters response.
+     *
+     * @param result the result returned as the response.
+     * @return the processed event result.
+     *//*
+    private static EventResult readCharactersResponse(EventResult<Boolean, EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        Boolean numberOfBytes = result.getResponse();
+        return result;
+    }
+    */
+
+    /**
+     * Reads characters asynchronously.
+     *
+     * @param numberOfCharacters number of characters which should be read.
+     * @param characterChannel   the channel which the characters will be read.
+     * @param context            context of the event.
+     * @return the content which is read.
+     * @throws ExecutionException   if an error occurs in the async framework.
+     * @throws InterruptedException during interrupt.
+     */
+    private String readCharacter(int numberOfCharacters, CharacterChannel characterChannel, EventContext context) throws
+            ExecutionException, InterruptedException {
+        ReadCharactersEvent event = new ReadCharactersEvent(characterChannel, numberOfCharacters, context);
+        CompletableFuture<EventResult> future = eventManager.publish(event);
+        EventResult eventResult = future.get();
+        Throwable error = ((EventContext) eventResult.getContext()).getError();
+        if (null != error) {
+            throw new ExecutionException(error);
+        }
+        return (String) eventResult.getResponse();
+    }
+
     /**
      * <p>
      * Reads characters from the channel.
      * </p>
-     *
+     * <p>
      * {@inheritDoc}
      */
     @Override
     public BValue[] execute(Context context) {
         BStruct channel;
         long numberOfCharacters;
-        BString content;
+        BString content = null;
+        BStruct errorStruct = null;
         try {
             channel = (BStruct) getRefArgument(context, CHAR_CHANNEL_INDEX);
             numberOfCharacters = getIntArgument(context, NUMBER_OF_CHARS_INDEX);
             CharacterChannel characterChannel = (CharacterChannel) channel.getNativeData(IOConstants
                     .CHARACTER_CHANNEL_NAME);
-            String readBytes = characterChannel.read((int) numberOfCharacters);
-            content = new BString(readBytes);
+            EventContext eventContext = new EventContext(context);
+            // IOUtils.read(characterChannel,numberOfCharacters,ReadCharacters::readCharactersResponse);
```

Comment:
Remove this commented line

Reply:
This needs to be uncommented upon the availability of the async functions